components/addContact/index.jsx

```diff
@@ -38,11 +38,33 @@ import { useRedirectIfLoggedOut } from "../../src/effects/auth";
 import styles from "./styles";
 
 const useStyles = makeStyles((theme) => createStyles(styles(theme)));
+export const EXISTING_WEBID_ERROR_MESSAGE =
+  "That WebID is already in your contacts";
+export const NO_NAME_ERROR_MESSAGE = "That WebID does not have a name";
 
-export function handleSubmit({ alertError, alertSuccess, fetch, webId }) {
+export function handleSubmit({
+  setIsLoading,
+  alertError,
+  alertSuccess,
+  fetch,
+  webId,
+}) {
   return async (iri) => {
+    setIsLoading(true);
     const { response: userProfile } = await getResource(webId, fetch);
     const pod = getUrl(userProfile.dataset, space.storage);
+    const addressBookIri = joinPath(pod, "contacts/");
```

Comment:
Use the hook `useAddressBook` instead, and use `getSourceUrl` from the Solid Client JS to get the url.